scipy/stats/_continuous_distns.py

```diff
@@ -6428,6 +6428,12 @@ def _munp(self, n, c):
 rdist = rdist_gen(a=-1.0, b=1.0, name="rdist")
 
 
+def _rayleigh_fit_check_error(ier, msg):
+    if ier != 1:
+        raise RuntimeError('rayleigh.fit failed to find the root of the MLE '
+                           f'equation: {msg} (ier={ier})')
```

Comment:
```suggestion
                           f'first-order condition: {msg} (ier={ier})')
```
or similar?

Reply:
It's tough to generate a case for the test suite, but I tested this locally by modifying the function to be solved such that it would not make progress. Looks good.
```
runfile('C:/Users/mhaberla/Dropbox/test_rayleigh.py', wdir='C:/Users/mhaberla/Dropbox')
Traceback (most recent call last):

  File "C:\Users\mhaberla\Dropbox\test_rayleigh.py", line 10, in <module>
    p = rayleigh.fit(data)

  File "C:\Users\mhaberla\Desktop\scipy\scipy\stats\_continuous_distns.py", line 6554, in fit
    _rayleigh_fit_check_error(ier, msg)

  File "C:\Users\mhaberla\Desktop\scipy\scipy\stats\_continuous_distns.py", line 6433, in _rayleigh_fit_check_error
    raise RuntimeError('rayleigh.fit failed to find the root of the MLE '

RuntimeError: rayleigh.fit failed to find the root of the MLE equation: The iteration is not making good progress, as measured by the 
  improvement from the last ten iterations. (ier=5)
```

Reply:
I updated the error message--maybe it's too much now?
```
RuntimeError: rayleigh.fit: fsolve failed to find the root of the first-order conditions of the log-likelihood function: The iteration is not making good progress, as measured by the 
  improvement from the last ten iterations. (ier=5)
```